internal/rpc/client/ospkg/client_test.go

```diff
@@ -0,0 +1,186 @@
+	type detectInput struct {
+		req *detector.OSDetectRequest
+	}
+	type detectOutput struct {
+		res *detector.DetectResponse
+		err error
+	}
+	type detect struct {
+		input  detectInput
+		output detectOutput
+	}
+
+	type fields struct {
+		token string
+	}
+	type args struct {
+		osFamily string
+		osName   string
+		pkgs     []analyzer.Package
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		detect  detect
+		want    []types.DetectedVulnerability
+		wantErr string
+	}{
+		{
+			name: "happy path",
+			fields: fields{
+				token: "token",
+			},
+			args: args{
+				osFamily: "alpine",
+				osName:   "3.10.2",
+				pkgs: []analyzer.Package{
+					{
+						Name:    "openssl",
+						Version: "1.0.1e",
+						Release: "1",
+						Epoch:   0,
+					},
+				},
+			},
+			detect: detect{
+				input: detectInput{
+					req: &detector.OSDetectRequest{
+						OsFamily: "alpine",
+						OsName:   "3.10.2",
+						Packages: []*detector.Package{
+							{
+								Name:    "openssl",
+								Version: "1.0.1e",
+								Release: "1",
+								Epoch:   0,
+							},
+						},
+					},
+				},
+				output: detectOutput{
+					res: &detector.DetectResponse{
+						Vulnerabilities: []*detector.Vulnerability{
+							{
+								VulnerabilityId:  "CVE-2019-0001",
+								PkgName:          "bash",
+								InstalledVersion: "1.2.3",
+								FixedVersion:     "1.2.4",
+								Title:            "RCE",
+								Description:      "Remote Code Execution",
+								Severity:         detector.Severity_HIGH,
+							},
+						},
+					},
+				},
+			},
+			want: []types.DetectedVulnerability{
+				{
+					VulnerabilityID:  "CVE-2019-0001",
+					PkgName:          "bash",
+					InstalledVersion: "1.2.3",
+					FixedVersion:     "1.2.4",
+					Vulnerability: dbTypes.Vulnerability{
+						Title:       "RCE",
+						Description: "Remote Code Execution",
+						Severity:    "HIGH",
+					},
+				},
+			},
+		},
+		{
+			name:   "Detect returns an error",
+			fields: fields{},
+			args: args{
+				osFamily: "alpine",
+				osName:   "3.10.2",
+				pkgs: []analyzer.Package{
+					{
+						Name:    "openssl",
+						Version: "1.0.1e",
+						Release: "1",
+						Epoch:   0,
+					},
+				},
+			},
+			detect: detect{
+				input: detectInput{
+					req: &detector.OSDetectRequest{
+						OsFamily: "alpine",
+						OsName:   "3.10.2",
+						Packages: []*detector.Package{
+							{
+								Name:    "openssl",
+								Version: "1.0.1e",
+								Release: "1",
+								Epoch:   0,
+							},
+						},
+					},
+				},
+				output: detectOutput{
+					err: xerrors.New("error"),
+				},
+			},
+			wantErr: "failed to detect vulnerabilities via RPC",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockDetector := new(mockDetector)
+			mockDetector.On("Detect", mock.Anything, tt.detect.input.req).Return(
+				tt.detect.output.res, tt.detect.output.err)
+			d := DetectClient{
```

Comment:
ditto

Reply:
I fixed it here.
https://github.com/aquasecurity/trivy/pull/303/commits/c421ba227cba1c5c1208dc97045e80ec6659909a#diff-9e840e732d4ed2683d55bf1704c6e543R171